ContextualPopupDecorator/ContextualPopup.js

```diff
@@ -0,0 +1,204 @@
+ * @ui
+ * @public
+ */
+const ContextualPopupBase = kind({
+	name: 'ContextualPopup',
+
+	propTypes: /** @lends agate/ContextualPopupDecorator.ContextualPopup.prototype */ {
+		/**
+		 * The contents of the popup.
+		 *
+		 * @type {Node}
+		 * @required
+		 * @public
+		 */
+		children: PropTypes.node.isRequired,
+
+		/**
+		 * Style object for arrow position.
+		 *
+		 * @type {Object}
+		 * @public
+		 */
+		arrowPosition: PropTypes.shape({
+			bottom: PropTypes.number,
+			left: PropTypes.number,
+			right: PropTypes.number,
+			top: PropTypes.number
+		}),
+
+		/**
+		 * Style object for container position.
+		 *
+		 * @type {Object}
+		 * @public
+		 */
+		containerPosition: PropTypes.shape({
+			bottom: PropTypes.number,
+			left: PropTypes.number,
+			right: PropTypes.number,
+			top: PropTypes.number
+		}),
+
+		/**
+		 * Called with the reference to the container node.
+		 *
+		 * @type {Object|Function}
+		 * @public
+		 */
+		containerRef: EnactPropTypes.ref,
+
+		/**
+		 * Direction of ContextualPopup.
+		 *
+		 * @type {('above'|'above center'|'above left'|'above right'|'below'|'below center'|'below left'|'below right'|'left middle'|'left top'|'left bottom'|'right middle'|'right top'|'right bottom')}
+		 * @default 'below'
+		 * @public
+		 */
+		direction: PropTypes.oneOf(['above', 'above center', 'above left', 'above right', 'below', 'below center', 'below left', 'below right', 'left middle', 'left top', 'left bottom', 'right middle', 'right top', 'right bottom']),
+
+		/**
+		 * Called when the close button is clicked.
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		onCloseButtonClick: PropTypes.func,
+
+		/**
+		 * Shows the close button.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		showCloseButton: PropTypes.bool
+	},
+
+	defaultProps: {
+		direction: 'below center',
+		showCloseButton: false
+	},
+
+	styles: {
+		css,
+		className: 'container'
+	},
+
+	computed: {
+		arrowDirection: ({direction}) => {
+			const [arrowDirection] = direction.split(' ');
+			return arrowDirection;
+		},
+		className: ({direction, showCloseButton, styler}) => styler.append(
+			{
+				fixedSize: direction === 'above' || direction === 'below'
+			},
+			direction.split(' '),
+			{reserveClose: showCloseButton}
+		),
+		closeButton: ({showCloseButton, onCloseButtonClick}) => {
+			if (showCloseButton) {
+				return (
+					<Button
+						aria-label={$L('Close')}
+						backgroundOpacity="transparent"
+						className={css.closeButton}
+						css={css}
+						icon="closex"
+						onTap={onCloseButtonClick}
+						size="small"
+					/>
+				);
+			}
+		}
+	},
+
+	render: ({arrowDirection, arrowPosition, children, className, closeButton, containerPosition, containerRef, direction, ...rest}) => {
+		delete rest.onCloseButtonClick;
+		delete rest.showCloseButton;
+
+		return (
+			<ContextualPopupRoot aria-live="off" role="alert" {...rest} className={css.contextualPopup} direction={direction}>
```

Comment:
Do we need the `direction={direction}>` here?

Reply:
I added that as I was receiving a lint warning that direction is defined but never used, although I added "delete rest.direction;". It was the only way I managed to fix the warning. Do you have other suggestions? Thanks.

Reply:
Unfortunately, for now it is the only way to fix the lint warning because we don't know which props could be passed from `ContextualPopupBase` to `ContextualPopupRoot` like `style`.

Reply:
Yes, but "delete rest.direction;" doesn't fix the warning in this case.

Reply:
Oh really? Let me check.